File: src/UavPathOptimization.Domain/Entities/UavEntities/UavModel.cs
﻿using UavPathOptimization.Domain.Entities.Base;
using UnitsNet;

namespace UavPathOptimization.Domain.Entities.UavEntities;

public class UavModel : Entity
{
    public string Name { get; set; } = null!;

    public Speed MaxSpeed { get; set; }

    public TimeSpan MaxFlightTime { get; set; }
}

File: src/UavPathOptimization.WebAPI/uav-optimizer/src/models/UavModel.ts
export class UavModel {
  id: string;
  name: string;
  maxSpeed: number;
  maxFlightTime: string;

  constructor(id: string, name: string, maxSpeed: number, maxFlightTime: string) {
    this.id = id;
    this.name = name;
    this.maxSpeed = maxSpeed;
    this.maxFlightTime = maxFlightTime;
  }
}
